src/board.rs

```diff
@@ -104,25 +105,37 @@ impl Board {
         {
             Some(Some(square)) => {
                 *square = Square::Occupied(player, value);
-                Ok(())
+                Ok(ChangeDetail {
+                    square: square.to_owned(),
+                    coordinate: position,
+                })
             }
             Some(None) => Err(GamePlayError::InvalidPosition { position }),
             None => Err(GamePlayError::OutSideBoardDimensions { position }),
         }
     }
 
     // TODO: safety on index access like get and set - ideally combine error checking for all 3
-    pub fn clear(&mut self, position: Coordinate) {
+    pub fn clear(&mut self, position: Coordinate) -> Option<ChangeDetail> {
         if let Some(pos) = self
             .squares
             .get_mut(position.y as usize)
             .and_then(|y| y.get_mut(position.x as usize))
         {
-            *pos = Some(Square::Empty);
+            match pos.replace(Square::Empty) {
```

Comment:
Useful func: [Option::replace](https://doc.rust-lang.org/std/option/enum.Option.html#method.replace)

Puts something in an option and returns what was previously there